core/src/main/java/org/apache/iceberg/view/BaseSQLViewRepresentation.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+@Value.Include(value = SQLViewRepresentation.class)
```

Comment:
@amogh-jahagirdar `@Value.Include` is what does the trick. Here's what the Javadoc says:
```
Includes specified abstract value types into generation of processing. This is usually used to generate immutable implementation of classes from different packages that source code cannot be changed to place @Value.Immutable. Only public types of suppored kinds is supported 
```

Reply:
Nice! Ok good there's a way then to not duplicate the class and retain backwards compatibility.